<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Title</title>
</head>
<body>
<script type="text/javascript">
  /*
    使用promise：有两个异步操作。等第一个异步操作执行完，再执行第二个异步操作
   */

  const promise = new Promise((resolve, reject) => {
    // 执行第一个异步操作
    setTimeout(() => {
      console.log('第一个异步操作执行完了~');
      const data = 111;
      resolve(data);
    }, 1000)
  });

  /*
    then / catch方法默认返回是一个成功状态的promise对象
    如果指定返回值且是promise，那就以promise为准

    如果上一个是失败的，遇到了then方法不会执行。会往下走，往下走状态还是失败的
    如果上一个是成功的，遇到了catch方法不会执行。会往下走，往下走状态还是成功的

    catch只会设置一个，一般放在最后
   */
  promise
    .then((result) => {
      console.log('成功了~', result);

      // 执行第二个异步操作
      return new Promise((resolve, reject) => {
        setTimeout(() => {
          console.log('第二个异步操作执行完了~');
          const data = 222;
          reject(data);
        }, 1000)
      })
    })
    .then((result) => {
      console.log('成功了~', result);
    })
    .catch((error) => {
      console.log('失败了2~', error);
    })


</script>
</body>
</html>